integration_tests/src/main/python/parquet_testing_test.py

```diff
@@ -73,42 +73,38 @@
     _error_files["lz4_raw_compressed.parquet"] = "Exception"
     _error_files["lz4_raw_compressed_larger.parquet"] = "Exception"
 
-def hdfs_glob(path, pattern):
+def hdfs_glob(path_str, pattern):
     """
     Finds hdfs files by checking the input path with glob pattern
 
-    :param path: hdfs path to check 
-    :type path: pathlib.Path 
+    :param path_str: hdfs path to check 
+    :type path_str: str
     :return: generator of matched files
     """
-    path_str = path.as_posix()
-    full_pattern = path_str + '/' + pattern
-    cmd = ['hadoop', 'fs', '-ls', '-C', full_pattern]
-
-    process = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, universal_newlines=True)
-    stdout, stderr = process.communicate()
-    if process.returncode != 0:
-        raise AssertionError(f'Failed to list files from {path_str}. Error: {stderr}')
-    
-    paths = stdout.strip().split('\n')
-
-    for p in paths:
-        yield Path(p)
-
-def glob(path, pattern):
+    from spark_init_internal import get_spark_i_know_what_i_am_doing
+    full_pattern = os.path.join(path_str, pattern)
+    sc = get_spark_i_know_what_i_am_doing().sparkContext
+    config = sc._jsc.hadoopConfiguration()
+    fs_path = sc._jvm.org.apache.hadoop.fs.Path(full_pattern)
+    fs = sc._jvm.org.apache.hadoop.fs.FileSystem.get(fs_path.toUri(), config)
+    statuses = fs.globStatus(fs_path)
+    for status in statuses:
+        yield status.getPath().toString()
+
+def glob(path_str, pattern):
     """
     Finds files by checking the input path with glob pattern.
     Support local file system and hdfs
 
-    :param path: input path to check 
-    :type path: pathlib.Path 
+    :param path_str: input path to check 
+    :type path_str: str
     :return: generator of matched files
     """
-    path_str = path.as_posix()
     if not path_str.startswith('hdfs:'):
```

Comment:
not for this PR, but this should be agnostic of a particular non-local FileSystem path.